### PR TITLE
Fix stackoverflow if a test fails

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/85/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/scala/centaur/test/Test.scala [hunk 1]
```diff
@@ -100,28 +98,37 @@ object Operations {
   }
 
   def validateMetadata(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata): Test[Unit] = {
-    val consistencyTimeout = Future { Thread.sleep(CentaurConfig.metadataConsistencyTimeout.toMillis) }
+
+    def eventually(startTime: OffsetDateTime, timeout: FiniteDuration)(f: => Try[Unit]): Try[Unit] = {
```
Comment (mcovarr) [inline src/main/scala/centaur/test/Test.scala:103]: The way this is written kind of looks tail recursive but it's not.  It's easy enough to revise this into a tail recursive `match` if that's important.

Comment (geoffjentry) [inline src/main/scala/centaur/test/Test.scala:103]: It just needs to be stack friendly. My original one was just basic recursive so it was not.

Comment (mcovarr) [inline src/main/scala/centaur/test/Test.scala:103]: OK this one is basic recursive too so it sounds like it needs to be fixed.

File: src/main/scala/centaur/test/Test.scala [hunk 2]
```diff
@@ -100,28 +98,37 @@ object Operations {
...
+
+      f recoverWith {
+        case _ if OffsetDateTime.now().isBefore(startTime.plusSeconds(timeout.toSeconds)) =>
+          Thread.sleep(1.second.toMillis)
+          eventually(startTime, timeout)(f)
+      }
+    }
 
     new Test[Unit] {
       def validateMetadataUntilTimeout(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata): Try[Unit] = {
```
Comment (mcovarr) [inline src/main/scala/centaur/test/Test.scala]: This method no longer iterates so the name doesn't really fit any more.

File: src/main/scala/centaur/test/Test.scala (hunk 1 above)
Comment (mcovarr) [inline src/main/scala/centaur/test/Test.scala:103]:
``` scala
    @tailrec
    def eventually(startTime: OffsetDateTime, timeout: FiniteDuration)(f: => Try[Unit]): Try[Unit] = {
      import scala.concurrent.duration._

      f match {
        case Failure(_) if OffsetDateTime.now().isBefore(startTime.plusSeconds(timeout.toSeconds)) =>
          Thread.sleep(1.second.toMillis)
          eventually(startTime, timeout)(f)
        case t => t
      }
    }
```

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/85/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

**Pull request merged**